metricbeat/module/http/json/json_integration_test.go

```diff
@@ -15,32 +15,53 @@ import (
 func TestFetch(t *testing.T) {
 	compose.EnsureUp(t, "http")
 
-	f := mbtest.NewEventFetcher(t, getConfig())
+	f := mbtest.NewEventsFetcher(t, getConfig("object"))
 	event, err := f.Fetch()
 	if !assert.NoError(t, err) {
 		t.FailNow()
 	}
 
 	t.Logf("%s/%s event: %+v", f.Module().Name(), f.Name(), event)
+
+	f = mbtest.NewEventsFetcher(t, getConfig("array"))
+	event, err = f.Fetch()
+	if !assert.NoError(t, err) {
+		t.FailNow()
+	}
+
+	t.Logf("%s/%s event: %+v", f.Module().Name(), f.Name(), event)
 }
 
 func TestData(t *testing.T) {
 	compose.EnsureUp(t, "http")
 
-	f := mbtest.NewEventFetcher(t, getConfig())
-	err := mbtest.WriteEvent(f, t)
+	f := mbtest.NewEventsFetcher(t, getConfig("object"))
+	err := mbtest.WriteEvents(f, t)
 	if err != nil {
 		t.Fatal("write", err)
 	}
+
 }
 
-func getConfig() map[string]interface{} {
+func getConfig(jsonType string) map[string]interface{} {
```

Comment:
you could make this a bool called `isArray`. But that also works. Do you expect other types here in the future?

Reply:
If confirming to JSON standards, yes. A string, number, object, array, true, false and null could be valid JSON. I would say to leave it this way than adding a boolean between array or not.

Reply:
Interesting point about the other types. I wonder when a request for that will pop up :-) Lets go with your version.